### ISSUE TITLE
Value's Type method makes it not a drop-in replacement

### ISSUE THREAD

**Issue by alercah** [spf13/pflag#228]
The `Type` method on [`Value`](https://github.com/spf13/pflag/blob/master/flag.go#L185-L191) means that you don't get drop-in replacement for the base `flag` package's [equivalent](https://godoc.org/flag#Value).

It's not hard to wrap, but this should be called out as a specific incompatibility in the package header. Additionally, the `Type` method should be properly documented.

**Comment by stevenh** [comment]
This just caught me out, and as you say there's no documentation on what type, so no idea what its meant to return.

Would be better if this could be made optional so it is a drop in replacement.

**Comment by AdamSLevy** [comment]
Using reflect a default type string could easily be generated. The fact that implementing Type() string is forced on the user of this package is extremely annoying.

**Comment by tomasaschan** [comment]
This issue has not been updated in a long time. Is it still relevant?

(If I receive no answer before the end of the summer, I'll close it. Feel free to re-open if you come here after that happens and it is relevant to you - but if you do, please also comment and specify what outcome you're interested in!)

**Comment by spectria-limina** [comment]
Yes, this remains an issue in the latest version of `pflag`. It continues to break the promise of being a drop-in replacement for `flag`. The ideal fix is to remove the `Type()` method so that `pflag` can actually be a drop-in replacement.
